src/app/hero-list/hero-list.component.ts

```diff
@@ -35,6 +36,7 @@ export class HeroListComponent implements OnInit{
   }
 
   handlePreviousPage(): void {
+    this.heroes = [];
```

Comment:
i dont think previous and next page should reset heros , its rather a side effect of fetching new page